executor/aggregate.go

```diff
@@ -51,15 +51,32 @@ type baseHashAggWorker struct {
 	aggFuncs     []aggfuncs.AggFunc
 	maxChunkSize int
 	stats        *AggWorkerStat
+
+	memTracker *memory.Tracker
+	BInMap     *int // incident B in Go map
 }
 
-func newBaseHashAggWorker(ctx sessionctx.Context, finishCh <-chan struct{}, aggFuncs []aggfuncs.AggFunc, maxChunkSize int) baseHashAggWorker {
-	return baseHashAggWorker{
+const (
+	// ref https://github.com/golang/go/blob/go1.15.6/src/reflect/type.go#L2162.
+	// defBucketMemoryUsage = bucketSize*(1+unsafe.Sizeof(string) + unsafe.Sizeof(slice))+2*ptrSize
+	defBucketMemoryUsage = 8*(1+16+24) + 16
+	// Maximum average load of a bucket that triggers growth is 6.5.
+	// Represent as loadFactorNum/loadFactorDen, to allow integer math.
```

Comment:
Why do we need `allow integer math`?

Reply:
If we use 6.5 directly, golang complier will throw an error `constant 6.5 truncated to integer` in the condition `len(mapper) > (1<<*w.BInMap)*loadFactorNum/loadFactorDen`.
If we want it run, maybe we need add some type conversions, eg `float64(len(mapper)) > float64(int(1<<*w.BInMap))*6.5`. I think it is inefficient.